package valhalla.mjolnir;

message Transit {

  message LatLng {
    optional float lat = 1;
    optional float lng = 2;
  }
  
  message Stop {
    optional LatLng ll = 1;
    optional uint64 graphid = 2;
    optional string name = 3;
    optional string onestop_id = 4;
    optional uint64 osm_way_id = 5;
    optional uint32 stop_desc = 6;
    optional string timezone = 7;
    optional bool wheelchair_boarding = 8;
    optional string zone_id = 9;
  }

  message StopPair {
    optional bool bikes_allowed = 1; 
    optional string block_id = 2; 
    optional string destination_arrival_time = 3;
    optional uint64 destination_graphid = 4;
    optional string destination_onestop_id = 5; 
    optional string operator_onestop_id = 6;
    optional string origin_departure_time = 7;
    optional uint64 origin_graphid = 8;
    optional string origin_onestop_id = 9;
    optional uint32 route_key = 10;
    optional string route_onestop_id = 11;
    repeated string service_added_dates = 12;
    repeated bool service_days_of_week = 13;
    optional string service_end_date = 14;
    repeated string service_except_dates = 15;
    optional string service_start_date = 16;
    optional string trip = 17;
    optional string trip_headsign = 18;
    optional uint32 trip_key = 19; 
    optional string trip_short_name = 20;
    optional bool wheelchair_accessible = 21;
  }
 
  enum VehicleType {
    kTram = 0;
    kSubway = 1;
    kRail = 2;
    kBus = 3;
    kFerry = 4;
    kCableCar = 5;
    kGondola = 6;
    kFenicular = 7;
  }
 
  message Route {
    optional uint32 key = 1;
    optional string name = 2;
    optional string onestop_id = 3;
    optional string operated_by_name = 4;
    optional string operated_by_onestop_id = 5;
    optional string route_color = 6;
    optional string route_desc = 7; 
    optional string route_long_name = 8;
    optional string route_text_color = 9;
    optional VehicleType vehicle_type = 10;     
  }
 
  repeated Stop stop = 1;
  repeated StopPair stop_pair = 2;
  repeated Route route = 3; 
}
